pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -42,17 +42,27 @@ func (b containersByID) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
 func (b containersByID) Less(i, j int) bool { return b[i].ID.ID < b[j].ID.ID }
 
 // Newest first.
-type podSandboxByCreated []*runtimeapi.PodSandbox
+type podSandboxByAttempt []*runtimeapi.PodSandbox
 
-func (p podSandboxByCreated) Len() int           { return len(p) }
-func (p podSandboxByCreated) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
-func (p podSandboxByCreated) Less(i, j int) bool { return p[i].CreatedAt > p[j].CreatedAt }
+func (p podSandboxByAttempt) Len() int      { return len(p) }
+func (p podSandboxByAttempt) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+func (p podSandboxByAttempt) Less(i, j int) bool {
+	if p[i].Metadata.Attempt == p[j].Metadata.Attempt {
+		return p[i].CreatedAt > p[j].CreatedAt
```

Comment:
should it be reverse?

Reply:
i think not.

Reply:
Now as the previous definition is fixed and doesn't require the call to `Reverse`, it makes sense.